lib/ansible/modules/system/xfs_quota.py

```diff
@@ -0,0 +1,367 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# Copyright: (c) 2018, Emmanouil Kampitakis <info@kampitakis.de>
+# Copyright: (c) 2018, William Leemans <willie@elaba.net>
+
+# Contributors:
+# (c) 2018, Emmanouil Kampitakis <info@kampitakis.de>
+# (c) 2018, Nicolas Magliaro <nicolasmagliaro@gmail.com>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+ANSIBLE_METADATA = {
+    'metadata_version': '1.1',
+    'status': ['preview'],
+    'supported_by': 'community'
+}
+
+DOCUMENTATION = '''
+---
+module: xfs_quota
+short_description: Set Quotas on XFS filesystems
+description:
+  - Configure Quotas on XFS filesystems. /etc/projects and /etc/projid needs to be configured before calling this module when setting project quotas.
+version_added: "2.8"
+author: "William Leemans (@bushvin) <willie@elaba.net>"
+options:
+  type:
+    description:
+      - The xfs quota type.
+    required: true
+    choices:
+      - user
+      - group
+      - project
+  name:
+    description:
+      - the name of the user, group or project to apply the quota to, if other than default
+  mountpoint:
+    description:
+      - the mountpoint on which to apply the quotas
+    required: true
+  bhard:
+    description:
+      - hard blocks quota limit
+  bsoft:
+    description:
+      - soft blocks quota limit
+  ihard:
+    description:
+      - hard inodes quota limit
+  isoft:
+    description:
+      - soft inodes quota limit
+  rtbhard:
+    description:
+      - hard realtime blocks quota limit
+  rtbsoft:
+    description:
+      - soft realtime blocks quota limit
+  state:
+    description:
+      - Whether to apply the limits or remove them.
+      - When removing limit, they are set to 0, and not quite removed.
+    default: present
+    choices:
+      - present
+      - absent
+
+notes:
+   - Not tested on any debian based system
+requirements:
+   - xfsprogs
+'''
+
+EXAMPLES = r'''
+- name: Set default project soft and hard limit on /opt of 1g
+  xfs_quota:
+    type: project
+    mountpoint: /opt
+    bsoft: 1g
+    bhard: 1g
+    state: present
+
+- name: Remove the default limits on /opt
+  xfs_quota:
+    type: project
+    mountpoint: /opt
+    state: absent
+
+- name: Set default soft user inode limits on /home of 1024 inodes and hard of 2048
+  xfs_quota:
+    type: user
+    mountpoint: /home
+    isoft: 1024
+    ihard: 2048
+
+'''
+
+RETURN = ''' # '''
+
+import os
+import sys
+import json
+import pwd
+import grp
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.basic import human_to_bytes
+
+
+def main():
+    module = AnsibleModule(
+        argument_spec=dict(
+            type=dict(required=True, choices=['user', 'group', 'project']),
```

Comment:
If the order is of no importance, always sort alphabetically. (That should have been part of the Zen of Python ;-))
Also, start every entry with the parameter type:
```suggestion
            type=dict(type='str', required=True, choices=['user', 'group', 'project']),
```

Reply:
done